accounting.js

```diff
@@ -0,0 +1,49 @@
+const log4js = require('log4js');
+const logger = log4js.getLogger('<filename>');
+log4js.configure({
+    appenders: {
+        file: { type: 'fileSync', filename: 'logs/debug.log' }
+    },
+    categories: {
+        default: { appenders: ['file'], level: 'debug'}
+    }
+});
+const moment = require('moment');
+
+function doAccounts(dataFile) {
+    let namesList = [];
+    for (let i = 0; i < dataFile.length; i++) {
+        namesList.push(dataFile[i].From);
+        namesList.push(dataFile[i].To);
+    }
+    let accountsList = Array.from(new Set(namesList));
+    
+    let accounts = [];
+    for (let i = 0; i < accountsList.length; i++) {
+        accounts[i] = {Name: accountsList[i], Net_Balance: 0, Transactions: []};
+    }

```

Comment:
Array has a `map` method that's useful for these sort of transformations ("for each thing in the array, I want another thing based on it"). In this case it'd look like:

    const accounts = accountsList.map(accountName => ({ Name: accountName, Net_Balance: 0, Transactions: [] }))

Also, you're mixing camelCase and snake_case (use camelCase), and you should start field names (Name, Net_Balance, Transactions) with lowercase.

Reply:
Having read on a bit, you're having to loop over your accounts to find the right one as you read the data file; a better approach when there's a single key value you want to use to look items up (in this case the account name) is to build a dictionary.